### PR TITLE
stm32wl55 : add more tests on the nucleo board

### PR DESCRIPTION
**Pull request by FRASTM**
this PR makes the test case with hardware connection pass on the nucleo_wl55jc target board from STMicrolectronics

- tests/drivers/uart/uart_async_api  --> connect USART1 D1 / TX (pb6) to D0 / RX (pb7) on the ARDuino connector CN9 of the nucleo board
- tests/drivers/spi/spi_loopback --> connect SPI1 MISO  pin D12 (pa6) on SPI1 MOSI pin D11 (pa7) on the ARDuino connector CN5 of the nucleo board

Each testcase includes an overlay for this board.
build for the target :
`west build -p auto -b nucleo_wl55jc tests/drivers/uart/uart_async_api/`
`west build -p auto -b nucleo_wl55jc tests/drivers/spi/spi_loopback/`

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]

File: tests/drivers/uart/uart_async_api/src/test_uart.h [hunk 1]
```diff
@@ -31,7 +31,8 @@
 #define UART_DEVICE_NAME DT_LABEL(DT_NODELABEL(sercom1))
 #elif defined(CONFIG_BOARD_NUCLEO_F103RB) || \
 	defined(CONFIG_BOARD_NUCLEO_G071RB) || \
-	defined(CONFIG_BOARD_NUCLEO_G474RE)
+	defined(CONFIG_BOARD_NUCLEO_G474RE) || \
```
Comment (erwango) [inline tests/drivers/uart/uart_async_api/src/test_uart.h:34]: Pure comment on  that part:
Isn't is possible to factorize a bit here and use `DT_NODELABEL(arduino_serial)` on a number of boards ?

File: boards/arm/nucleo_wl55jc/arduino_r3_connector.dtsi [hunk 2]
```diff
@@ -35,6 +35,6 @@
 	};
 };
 
-arduino_serial: &lpuart1 {};
+arduino_serial: &usart1 {}; /* &lpuart1 {}; */
```
[the remark this answers is not in the document]
Comment (FRASTM) [reply boards/arm/nucleo_wl55jc/arduino_r3_connector.dtsi]: comment removed

File: tests/drivers/uart/uart_async_api/src/test_uart.h (hunk 1 above)
Comment (FRASTM) [reply tests/drivers/uart/uart_async_api/src/test_uart.h:34]: sort the uart instance in the `test_uart.h` depending on the stm32 target board

Comment (erwango) [reply tests/drivers/uart/uart_async_api/src/test_uart.h:34]: ok, that's a bit better.

**Review by erwango — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]

**Pull request merged**